File: VB/RepositoryItemPopupContainerEditOKButton.vb
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Drawing
Imports System.ComponentModel

Namespace PopupContainerControlWithOKButton

    Friend Class RepositoryItemPopupContainerEditOKButton
        Inherits RepositoryItemPopupContainerEdit

        ' static constructor which calls static registration method
        Shared Sub New()
            Call RegisterPopupContainerEditOKButton()
        End Sub

        ' static register method
        Public Shared Sub RegisterPopupContainerEditOKButton()
            Call EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(PopupContainerEditOKButtonEditorName, GetType(PopupContainerEditOKButton), GetType(RepositoryItemPopupContainerEditOKButton), GetType(PopupContainerEditViewInfo), New ButtonEditPainter(), True, CType(Nothing, System.Drawing.Image)))
        End Sub

        ' internal editor name
        Friend Const PopupContainerEditOKButtonEditorName As String = "PopupContainerEditOKButton"

        ' public constructor
        Public Sub New()
            MyBase.New()
            protShowOkButton = True
        End Sub

        Protected protShowOkButton As Boolean

        <Description("Determines whether the Ok button will be displayed in the popup form or not.")>
        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Property ShowOkButton As Boolean
            Get
                Return protShowOkButton
            End Get

            Set(ByVal value As Boolean)
                If value <> protShowOkButton Then protShowOkButton = value
            End Set
        End Property

        ' ovverride property
        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return PopupContainerEditOKButtonEditorName
            End Get
        End Property

        Public Overrides Sub Assign(ByVal item As RepositoryItem)
            MyBase.Assign(item)
            Dim currentRepository As RepositoryItemPopupContainerEditOKButton = TryCast(item, RepositoryItemPopupContainerEditOKButton)
            ShowOkButton = currentRepository.ShowOkButton
        End Sub
    End Class
End Namespace

File: CS/RepositoryItemPopupContainerEditOKButton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Drawing;
using System.ComponentModel;

namespace PopupContainerControlWithOKButton
{
    class RepositoryItemPopupContainerEditOKButton : RepositoryItemPopupContainerEdit
    {
        // static constructor which calls static registration method
        static RepositoryItemPopupContainerEditOKButton() { RegisterPopupContainerEditOKButton(); }

        // static register method
        public static void RegisterPopupContainerEditOKButton()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(
                PopupContainerEditOKButtonEditorName,
                typeof(PopupContainerEditOKButton),
                typeof(RepositoryItemPopupContainerEditOKButton),
                typeof(PopupContainerEditViewInfo),
                new ButtonEditPainter(),
                true,
                null));
        }

        // internal editor name
        internal const string PopupContainerEditOKButtonEditorName = "PopupContainerEditOKButton";

        // public constructor
        public RepositoryItemPopupContainerEditOKButton() : base()
        {
            protShowOkButton = true;
        }

        protected bool protShowOkButton;
        [Description("Determines whether the Ok button will be displayed in the popup form or not.")]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public bool ShowOkButton
        {
            get { return protShowOkButton; }
            set { if (value != protShowOkButton) protShowOkButton = value; }
        }

        // ovverride property
        public override string EditorTypeName
        {
            get { return PopupContainerEditOKButtonEditorName; }
        }

        public override void Assign(RepositoryItem item)
        {
            base.Assign(item);
            RepositoryItemPopupContainerEditOKButton currentRepository = (item as RepositoryItemPopupContainerEditOKButton);
            this.ShowOkButton = currentRepository.ShowOkButton;
        }
    }
}
